protocol-units/bridge/contracts/src/AtomicBridgeCounterparty.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.22;
+
+import {Initializable} from "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+import {IAtomicBridgeCounterparty} from "./IAtomicBridgeCounterparty.sol";
+import {AtomicBridgeInitiator} from "./AtomicBridgeInitator.sol";
+
+contract AtomicBridgeCounterparty is IAtomicBridgeCounterparty, Initializable {
+    enum MessageState {
+        PENDING,
+        COMPLETED,
+        REFUNDED
+    }
+
+    struct BridgeTransferDetails {
+        bytes32 initiator; // movement address 
+        address recipient;
+        uint256 amount;
+        bytes32 hashLock;
+        uint256 timeLock;
+        MessageState state; 
+    }
+
+    AtomicBridgeInitiator public atomicBridgeInitiator;
+    mapping(bytes32 => BridgeTransferDetails) public bridgeTransfers; 
+
+    function initialize(address _atomicBridgeInitiator) public initializer {
+        if (_atomicBridgeInitiator == address(0)) revert ZeroAddress();
+        atomicBridgeInitiator = AtomicBridgeInitiator(_atomicBridgeInitiator);
+    }
+
+    function lockBridgeTransferAssets(
+        bytes32 initiator,
+        bytes32 bridgeTransferId,
+        bytes32 hashLock,
+        uint256 timeLock,
+        address recipient,
+        uint256 amount
+    ) external returns (bool) {
+        BridgeTransferDetails storage transfer = bridgeTransfers[bridgeTransferId];
+        if (transfer.recipient != address(0)) revert BridgeTransferInvalid();
```

Comment:
This is input validation, it checks that the address is not `0x0000000000000000000000000000000000000000` as this would be a valid values for `address`. If the address is 0x, then we revert with the Error. Updated now to 
`        if (recipient != address(0)) revert BridgeTransferInvalid();` to check the arg, not the storage val.

Reply:
Sorry correction, it means, we are expecting the `transfer.recipient` to be `address(0)` at this point as its initialised. If it's anything other than 0 then we should revert.

Reply:
It is _technically_ `0x` at this point in the program, that is, when it is uninitialised, before it is set further down with the `recipient` arg, solidity does this by default.  (If it wasn't `0x` the tests would not pass). But we can remove this check as I guess its fairly redundant anyway and doesn't provide much value.

Reply:
I'm sorry, Richard. It will always hold the address of the recipient because the bridge receives the recipient's address from the initiating event on the other side. It will never be 0x, see the abstract bridge code.

Reply:
The `recipient` will never be `0x` yes, but this check doesn't check `recipient` its checking against `transfer.recipient` which is the solidity struct field. Which is `0x` here.